src/utils/graph/constraints.js

```diff
@@ -11,96 +11,145 @@ import { distance1d, greaterOrEqual, equalTo, subtract } from './common';
  */
 export const rowConstraint = {
   property: 'y',
-  difference: subtract,
-  distance: distance1d,
-  operator: greaterOrEqual,
-  target: (a, b, co, constants) => constants.spaceY,
-  strength: () => 1,
-  weightA: () => 0,
-  weightB: () => 1,
-  required: true
+
+  solve: (constraint, constants) => {
```

Comment:
I know that the `solve` and `strict` methods are defined in multiple places, but I couldn't find any explanation anywhere about what it is they do. Do you mind adding some (JSDoc?) comments next to them explaining what they're for, and maybe what's different about each one and why?

Reply:
Ah I found them in solver.js - it just wasn't included in this PR. It might be worth mentioning where it's used in the comments still

Reply:
Yeah so at the moment that documentation lives in the solver, I'd definitely like to start adding more details here but this API is currently in a bit of flux with more changes coming. 

Once that settles I think then would be best to further document, if that sounds reasonable?

Reply:
sure thing